src/QueryCreator.tsx

```diff
@@ -87,12 +96,25 @@ class QueryCreator extends React.PureComponent<Props> {
     );
 
     sagaMiddleware.run(rootSaga);
+
     this.store.dispatch(appStart());
 
+    if (this.props.defaultTimezoneForQuery) {
```

Comment:
What do You think about get rid of 3 action dispatches and use `preloadedState` to set a default timezone in query and other settings ?
```
const preloadedState = rehydrateState();

const initialState = setConfiguration(preloadedState, defaultTimezoneForQuery, disableTimezoneSelection);

  this.store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(sagaMiddleware)
  );
```

Especially that store creator already accepts `initialState` argument.